core/src/main/java/ai/timefold/solver/core/api/solver/SolutionManager.java

```diff
@@ -84,6 +86,28 @@ public interface SolutionManager<Solution_, Score_ extends Score<Score_>> {
     @Nullable
     Score_ update(@NonNull Solution_ solution, @NonNull SolutionUpdatePolicy solutionUpdatePolicy);
 
+    /**
+     * This method updates all shadow variables at the entity level,
+     * simplifying the requirements of {@link SolutionManager#update(Object)}.
+     * Unlike the latter method,
+     * it does not require the complete configuration necessary to obtain an instance of {@link SolutionManager}.
+     * <p>
+     * However, this method requires that the entity does not define any shadow variables that rely on listeners,
+     * as it requires a complete solution.
+     *
+     * @param solutionClass the solution class
+     * @param entities all entities to be updated
+     */
+    static <Solution_> void updateShadowVariables(@NonNull Class<Solution_> solutionClass,
```

Comment:
Arguably we want the same, but for an entire solution.
That one can and should run the full update.